### PR TITLE
Documentation tidying

### PR DESCRIPTION
**Pull request by AWhetter**
This pull request includes many fixes to the documentation, including:
fixing sphinx syntax, fixing spelling mistakes, making some docstrings easier to understand, splitting notes into note blocks, code into code blocks, and examples into example blocks.

P.S. Apologies for making most of the changes in a single commit. It was just easier to do everything in one go.

### CODE REVIEW

**Comment by sqlboy** [comment]
Wow, this is great, thank you.

File: docs/conf.py [hunk 1]
```diff
@@ -108,7 +108,7 @@
 
 # The theme to use for HTML and HTML Help pages.  See the documentation for
 # a list of builtin themes.
-html_theme = 'default'
+html_theme = 'classic'
```
Comment (AWhetter) [inline docs/conf.py]: I've just noticed that this change has forced the template on readthedocs to use the classic template instead of the readthedocs template. How do you want to handle this?
If this is switched back to default it creates a warning when the docs are built. If we switch it to the readthedocs template then it requires a dependency on the `sphinx_rtd_theme` module.

Comment (justinfx) [inline docs/conf.py]: I would be fine with the warning as long as it allows certain build
environments to inject their preferred theme (ie readthedocs)

Comment (AWhetter) [inline docs/conf.py]: Cool. I've submitted a new pull request (#36) to revert the change

Comment (justinfx) [inline docs/conf.py]: Perfect. Thanks!
